Source/WebKit/NetworkProcess/NetworkResourceLoader.cpp

```diff
@@ -627,10 +627,10 @@ bool NetworkResourceLoader::shouldInterruptLoadForXFrameOptions(const String& xF
     case XFrameOptionsDisposition::SameOrigin: {
         auto origin = SecurityOrigin::create(url);
         auto topFrameOrigin = m_parameters.frameAncestorOrigins.last();
-        if (!origin->isSameSchemeHostPort(*topFrameOrigin))
+        if (topFrameOrigin && !origin->isSameSchemeHostPort(*topFrameOrigin))
```

Comment:
Is this still a valid check? e.g. will this still prevent a cross-origin website to embed an i-frame with `XFrameOptionsDisposition::SameOrigin`?

If topFrameOrigin is cross-origin, then it'll be nullptr here I think, but we won't ever return true in this function since the cross-origin ancestors are nullptr and we require both not nullptr and `!origin->isSameSchemeHostPort(*topFrameOrigin)` to return true here.